File: bank-account/src/main/java/org/suggs/sandbox/eventsourcing/bankaccount/accounts/commandhandler/ConfirmAccountToClientHandler.java
package org.suggs.sandbox.eventsourcing.bankaccount.accounts.commandhandler;

import com.google.common.eventbus.AllowConcurrentEvents;
import com.google.common.eventbus.Subscribe;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.suggs.sandbox.eventsourcing.bankaccount.accounts.domain.command.ConfirmAccountToClient;
import org.suggs.sandbox.eventsourcing.bankaccount.accounts.domain.event.AccountConfirmedToClient;
import org.suggs.sandbox.eventsourcing.bankaccount.accounts.repository.EventRepository;

import java.util.UUID;

public class ConfirmAccountToClientHandler implements CommandHandler<ConfirmAccountToClient> {

    private static final Logger LOG = LoggerFactory.getLogger(ConfirmAccountToClientHandler.class);
    private EventRepository eventRepository;

    public ConfirmAccountToClientHandler(EventRepository eventRepository) {
        this.eventRepository = eventRepository;
    }

    @Override
    @Subscribe
    @AllowConcurrentEvents
    public void handle(ConfirmAccountToClient aCommand) {
        confirmNewAccountNumberToClient(aCommand.getRequestId(), aCommand.getAccountNumber());
        eventRepository.save(AccountConfirmedToClient.Companion.anAccountConfirmedToClient(aCommand.getRequestId(), aCommand.getAccountNumber()));
    }

    private void confirmNewAccountNumberToClient(UUID requestId, UUID accountNumber) {
        LOG.debug("Emailing the client that we have finished setting up the new account for them (yay!!!");
    }
}

File: bank-account/src/main/kotlin/org/suggs/sandbox/eventsourcing/bankaccount/accounts/commandhandler/ConfirmAccountToClientHandler.kt
package org.suggs.sandbox.eventsourcing.bankaccount.accounts.commandhandler

import com.google.common.eventbus.AllowConcurrentEvents
import com.google.common.eventbus.Subscribe
import org.slf4j.LoggerFactory
import org.suggs.sandbox.eventsourcing.bankaccount.accounts.domain.command.ConfirmAccountToClient
import org.suggs.sandbox.eventsourcing.bankaccount.accounts.domain.event.AccountConfirmedToClient.Companion.accountConfirmedToClient
import org.suggs.sandbox.eventsourcing.bankaccount.accounts.repository.EventRepository
import java.util.*

class ConfirmAccountToClientHandler(private val eventRepo: EventRepository) : CommandHandler<ConfirmAccountToClient> {

    companion object{
        private val log = LoggerFactory.getLogger(this::class.java)
    }

    @Subscribe
    @AllowConcurrentEvents
    override fun handle(command: ConfirmAccountToClient) {
        confirmAccountToClient(command.accountNumber)
        eventRepo.save(accountConfirmedToClient(command.requestId, command.accountNumber))
    }

    private fun confirmAccountToClient(accountNumber: UUID) {
        log.info("Emailing client to inform that we have created a new account [$accountNumber] for them")
    }
}